os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -156,59 +134,89 @@ static int binary_manager_load(int bin_idx)
-
-	latest_ver = -1;
-	latest_idx = -1;
-	valid_bin_count = 0;
-
-	/* Open a directory for user binaries, BINARY_DIR_PATH */
-	dirp = opendir(BINARY_DIR_PATH);
-	if (!dirp) {
-		bmdbg("Failed to open a directory, %s\n", BINARY_DIR_PATH);
-		return ERROR;
-	}
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	binp = BIN_LOADINFO(bin_idx);
+	if (binp) {
+		valid_bin_count = 1;
+		latest_idx = 0;
+		snprintf(filepath[latest_idx], CONFIG_PATH_MAX, "%s/%s_%s", BINARY_DIR_PATH, BIN_NAME(bin_idx), BIN_VER(bin_idx));
+	} else
+#endif
+	{
+		bin_name = BIN_NAME(bin_idx);
+		name_len = strlen(bin_name);
+
+		latest_ver = -1;
+		latest_idx = -1;
+		valid_bin_count = 0;
+
+		/* Open a directory for user binaries, BINARY_DIR_PATH */
+		dirp = opendir(BINARY_DIR_PATH);
+		if (!dirp) {
+			bmdbg("Failed to open a directory, %s\n", BINARY_DIR_PATH);
+			return ERROR;
+		}
 
-	file_idx = 0;
+		file_idx = 0;
 
-	bmvdbg("Open a directory, %s\n", BINARY_DIR_PATH);
+		bmvdbg("Open a directory, %s\n", BINARY_DIR_PATH);
 
-	/* Read each directory entry */
-	for (;;) {
-		struct dirent *entryp = readdir(dirp);
-		if (!entryp) {
-			/* Finished with this directory */
-			break;
-		}
-		/* Read and Verify a binary file */
-		if (DIRENT_ISFILE(entryp->d_type) \
-			&& !strncmp(entryp->d_name, bin_name, name_len) && entryp->d_name[name_len] == '_') {
-			snprintf(filepath[file_idx], CONFIG_PATH_MAX, "%s/%s", BINARY_DIR_PATH, entryp->d_name);
-			ret = binary_manager_read_header(filepath[file_idx], &header_data[file_idx]);
-			if (ret == OK) {
-				valid_bin_count++;
-				version = (int)atoi(header_data[file_idx].bin_ver);
-				bmvdbg("Found valid header in version %d\n", version);
-				if (version > latest_ver) {
-					latest_ver = version;
-					latest_idx = file_idx;
-					bmvdbg("Latest version %d, idx %d\n", latest_ver, latest_idx);
+		/* Read each directory entry */
+		for (;;) {
+			struct dirent *entryp = readdir(dirp);
+			if (!entryp) {
+				/* Finished with this directory */
+				break;
+			}
+			/* Read and Verify a binary file */
+			if (DIRENT_ISFILE(entryp->d_type) \
+				&& !strncmp(entryp->d_name, bin_name, name_len) && entryp->d_name[name_len] == '_') {
+				snprintf(filepath[file_idx], CONFIG_PATH_MAX, "%s/%s", BINARY_DIR_PATH, entryp->d_name);
+				ret = binary_manager_read_header(filepath[file_idx], &header_data[file_idx]);
+				if (ret == OK) {
+					valid_bin_count++;
+					version = (int)atoi(header_data[file_idx].bin_ver);
+					bmvdbg("Found valid header in version %d\n", version);
+					if (version > latest_ver) {
+						latest_ver = version;
+						latest_idx = file_idx;
+						bmvdbg("Latest version %d, idx %d\n", latest_ver, latest_idx);
+					}
 				}
+				file_idx++;
 			}
-			file_idx++;
 		}
-	}
-	closedir(dirp);
+		closedir(dirp);
 
-	if (valid_bin_count == 0) {
-		bmdbg("No valid binary in fs : %s\n", BIN_NAME(bin_idx));
-		return ERROR;
+		if (valid_bin_count == 0) {
+			bmdbg("No valid binary in fs : %s\n", BIN_NAME(bin_idx));
```

Comment:
errcode is not necessary?

Reply:
Yes. currently this function doesn't distinguish error code because this result is not returned to user.